packages/x-date-pickers/src/AdapterMoment/AdapterMoment.ts

```diff
@@ -522,14 +522,25 @@ export class AdapterMoment implements MuiPickersAdapter<string> {
 
     let count = 0;
     let current = start;
+    let currentDayOfYear = current.get('dayOfYear');
     const nestedWeeks: Moment[][] = [];
 
     while (current.isBefore(end)) {
       const weekNumber = Math.floor(count / 7);
       nestedWeeks[weekNumber] = nestedWeeks[weekNumber] || [];
       nestedWeeks[weekNumber].push(current);
 
+      const prevDayOfYear = currentDayOfYear;
       current = this.addDays(current, 1);
+      currentDayOfYear = current.get('dayOfYear');
```

Comment:
The original snippet uses `.diff` but it also work if you just check the `dayOfYear` property which is a lot cheaper to access.

Reply:
Nice, this is a way more efficient and easier to understand solution.
Great job. 👍